scipy/special/boost_special_functions.h

```diff
@@ -951,26 +951,46 @@ ncf_kurtosis_excess_double(double v1, double v2, double l)
 
 template<typename Real>
 Real
-nct_cdf_wrap(const Real x, const Real v, const Real l)
+nct_cdf_wrap(const Real v, const Real l, const Real x)
 {
-    if (std::isfinite(x)) {
-        return boost::math::cdf(
-            boost::math::non_central_t_distribution<Real, StatsPolicy>(v, l), x);
+    if (std::isnan(x) || std::isnan(v) || std::isnan(l)) {
+	return NAN;
     }
-    // -inf => 0, inf => 1
-    return 1.0 - std::signbit(x);
+    if (v <= 0) {
+	sf_error("nctdtr", SF_ERROR_DOMAIN, NULL);
+	return NAN;
+    }
+    if (std::isinf(x)) {
+	return 1.0;
+    }
+    Real y;
+    try {
+	y = boost::math::cdf(
+                boost::math::non_central_t_distribution<Real, SpecialPolicy>(v, l), x);
+    } catch (...) {
+	/* Boost was unable to produce a result. */
+        sf_error("nctdtr", SF_ERROR_NO_RESULT, NULL);
+        y = NAN;
+    }
+    if ((y < 0) || (y > 1)) {
```

Comment:
I think this is a reasonable check. I wonder if you could find some input to trigger this branch of the code, if only for coverage purpose?

An alternative “trick” is to clamp the return value within $[0,1]$, which I see being used occasionally. Of course this relies on the assumption that the return value is out of range due to (small) numerical noise instead of wild algorithmic error.

Reply:
We don’t clamp because we’ve observed Boost returning complete nonsense far out of range in some cases. See the discussion for #21505.

Reply:
I did not stress test this function as much as @steppi did with the noncentral F distribution. I am quite sure though that there must be dark corners as the NCT and NCF distributions are based on the same non central beta in boost. Both of them also heavily rely on the incomplete beta function which can go bananas for extreme input.